remoting/getty/getty_client.go

```diff
@@ -204,6 +204,12 @@ func (c *Client) Request(request *remoting.Request, timeout time.Duration, respo
 	return perrors.WithStack(err)
 }
 
+// isAvailable returns true if the connection is available, or it can be re-established.
+func (c *Client) IsAvailable() bool {
+	_, _, err := c.selectSession(c.addr)
+	return err == nil
```

Comment:
```suggestion
	c, _, err := c.selectSession(c.addr)
	return c!=nil && err == nil
```

Reply:
add this defensive check, pls. check again.